cocotb/utils.py

```diff
@@ -630,3 +614,134 @@ def extract_coro_stack(coro, limit=None):
+    _ActivehdlVersion("10.5a.12.6914") > _ActivehdlVersion("10.5.216.6767")
+    """
+    pass
+
+
+class _CvcVersion(LooseVersion):
+    """
+    Version numbering class for Tachyon DA CVC.
+
+    Example:
+        >>> _CvcVersion("OSS_CVC_7.00b-x86_64-rhel6x of 07/07/14 (Linux-elf)") > _CvcVersion("OSS_CVC_7.00a-x86_64-rhel6x of 07/07/14 (Linux-elf)")
+        True
+    """
+    pass
+
+
+class _GhdlVersion(LooseVersion):
+    """Version numbering class for GHDL."""
+    pass
+
+
+class _IcarusVersion(LooseVersion):
+    """
+    Version numbering class for Icarus Verilog.
+
+    Example:
+        >>> _IcarusVersion("11.0 (devel)") > _IcarusVersion("10.3 (stable)")
+        True
+        >>> _IcarusVersion("10.3 (stable)") <= _IcarusVersion("10.3 (stable)")
+        True
+    """
+    pass
+
+
+class _ModelsimVersion(LooseVersion):
+    """Version numbering class for Mentor ModelSim."""
+    pass
+
+
+class _QuestaVersion(LooseVersion):
+    """
+    Version numbering class for Mentor Questa.
+
+    Example:
+        >>> _QuestaVersion("10.7c 2018.08") > _QuestaVersion("10.7b 2018.06")
+        True
+        >>> _QuestaVersion("2020.1 2020.01") > _QuestaVersion("10.7c 2018.08")
+        True
+    """
+    pass
+
+
+class _RivieraVersion(LooseVersion):
+    """
+    Version numbering class for Aldec Riviera-PRO.
+
+    Example:
+        >>> _RivieraVersion("2019.10.138.7537") == _RivieraVersion("2019.10.138.7537")
+        True
+    """
+    pass
+
+
+class _VcsVersion(LooseVersion):
+    """
+    Version numbering class for Synopsys VCS.
+
+    Example:
+        >>> _VcsVersion("Q-2020.03-1_Full64") > _VcsVersion("K-2015.09_Full64")
+        True
+    """
+    pass
+
+
+class _VerilatorVersion(LooseVersion):
+    """
+    Version numbering class for Verilator.
+
+    Example:
+        >>> _VerilatorVersion("4.032 2020-04-04") > _VerilatorVersion("4.031 devel")
+        True
+    """
+    pass
+
+
+class _XceliumVersion(LooseVersion):
+    """
+    Version numbering class for Cadence Xcelium.
+
+    Example:
+        >>> _XceliumVersion("20.06-g183") > _XceliumVersion("20.03-s002")
+        True
+        >>> _XceliumVersion("20.07-e501") > _XceliumVersion("20.06-g183")
+        True
+    """
+    pass
+
+
+class _IusVersion(_XceliumVersion):  # inherit everything from Xcelium
+    """
+    Version numbering class for Cadence IUS.
+
+    Example:
+        >>> _IusVersion("15.20-s050") > _IusVersion("15.20-s049")
+        True
+    """
+    pass
+
+
+if __name__ == "__main__":
+    import random
+    a = ""
+    for char in range(random.randint(250, 500)):
+        a += chr(random.randint(0, 255))
+    b = a
+    for error in range(random.randint(2, 9)):
+        offset = random.randint(0, len(a))
+        b = b[:offset] + chr(random.randint(0, 255)) + b[offset+1:]
+
+    diff = hexdiffs(a.encode('latin1'), b.encode('latin1'))
+    print(diff)
+
+    space = '\n' + (" " * 20)
+    print(space.join(diff.split('\n')))
```

Comment:
???

Reply:
Oh, I see this just moved from above. I think we can kill this with prejudice, or at least migrate it to a test.

Reply:
Please do.

Reply:
Let's do that  in a different PR.